multi-vm-chef-template-ubuntu-vm/azuredeploy.json

```diff
@@ -1,291 +1,312 @@
+            ]
         },
-        "protectedSettings": {
-          "validation_key": "[parameters('validation_key')]"
+        {
+            "type": "Microsoft.Compute/virtualMachines/extensions",
+            "name": "[concat(variables('vmName'),copyIndex(),'/', variables('vmExtensionName'))]",
+            "apiVersion": "2015-05-01-preview",
+            "location": "[resourceGroup().location]",
+            "copy": {
+                "name": "extensionLoop",
+                "count": "[parameters('numberOfInstances')]"
+            },
+            "properties": {
+                "publisher": "Chef.Bootstrap.WindowsAzure",
+                "type": "LinuxChefClient",
+                "typeHandlerVersion": "1210.12",
+                "settings": {
+                    "bootstrap_options": {
+                        "chef_node_name": "[concat(parameters('chef_node_name'),copyIndex())]",
+                        "chef_server_url": "[parameters('chef_server_url')]",
+                        "validation_client_name": "[parameters('validation_client_name')]",
+                        "node_ssl_verify_mode": "none"
+                    },
+                    "runlist": "[parameters('runlist')]",
+                    "validation_key_format": "plaintext"
+                },
+                "protectedSettings": {
+                    "validation_key": "-----BEGIN RSA PRIVATE KEY-----\nMI-------Rzf0Igf59e6i2yTYnKOyVxJdb2b9zkJtZjpRQbt46fmSQ\nElk9rc---------ntJHc1CQXVfQYQZVGccVAuPqNKJHmb70qJEi\nA1zvZudke3pj0Xkw/xAy1MnQBVTRtf23hIzZK4sraEyt\noQ93GwjpunSFJfI1GZKXDdsp/nRJi0qVU-------------9UOPO4G0PyPr+mcF\nxclZYUJMENgGIBFzhYVft0p7ezRinxkbnEW22FLxDplvBmfaYDTJZ6dmAVh6IHiM\naKtv5kj63-------------------------BAoIBAArCs5nsl6WcLtU2\n8Ci2K-----bghgcheAKBROCZLyDS\nGvNmvCc87jo6i8vGGvYOeic3ns/UraRp04LADQzz4E2vH6XVr/j35NqHK3HgvMtU\n1gpBbGpSrnMu8euTIxlGQSbfyY0+YdMYH29ZECVWtcwwLJE0E4MSTNQfqoGvKilk\nqySrsc9xlTrl7WemfM4jzYle9MlvZiG5enpyEriTY4xmyqHQE3A7q1AW9ry+aX/o\nyhre4Twt/q9Hon2BhBhCofW7J1laku+sLtUt3NEHqFo9W1qluN6zdLYQMeTNdHOn\ndUR7XQECgYEA6jSwp6dRZija45ytFEr7/vglpCG4DsfsX3WQdT8WbStpWK6qtlVI\nglZ4R+vgnupqxfLuy2ZCm+iIhJ8B3dGUVtlM5/HpUkyADCLt5oFbXzaK7NtHROjU\nxbh3NpwD3TmZn/6RRo3Vg39sqSOCr3+TbFtDAOCw3ZrEr7UFo0eblZECgYEA6HXa\nNPR0TPpMC3N5kD9GIwznjXJt8WuEOHp88LsIH3oEhjafb+kO+t2+VWztgCXFObS9\nYoYVoIxTjgXh93f364x7g4Y1F5pjkPqaONZI1Cc+r31BWjN/z6eHq3jK+mGw7vUx\nz3ogMQrMxa+v2wMJOkWGzAaiRCnw5aHUyKk9XiUCgYEA0ldODNSFk5W/uMpe26nY\nlo5w5+7sqkG9taIVgmeNkbB6ZY5BfY4sl/pm/lkHHFJ2YJ+k/YIy8mkuKsXP3FM8\nwZ9ZpbqrDSHtonjAZtPopyOKM91gLbBaX9UjJCWMhuQTaBEukkoR547KnUFqPon6\nckeobqJd1------------KfGMzVfMA91qCv4\nS/oKT3Ke3UgTkGm4mMTkeRtDMcT8UhJUv+SwFSncFzMehB8GvzzCFnLK+U6rrQvc\n/w+ZhU/5sr7n4U/MgeSEQrhBHH0ccHGaJXMsnaVSp23HnNxVaMNEPlestBaDdWsE\nxHVrzQKBgGEgnD9zXroS9thzDH3yYgxXIFnXyG35C9dy/05elWp7VT6ayqW714pB\nEHcpXZCHBdCgHh+D5QDPjmq+CoqBO5f/rv------rt\n5fCZX//M6qom30zFm85------cvmd\n-----END RSA PRIVATE KEY-----
+                }
```

Comment:
@suvadip30 , I got your issue. When `validation_key` is being passed from the parameters in `plaintext` format, all the new lines need to be replaced by `\n` and a single line string needs to be passed. This was not very user friendly and that's why we had added support to pass `validation_key` in `base64encoded` format as you pointed  https://github.com/Azure/azure-quickstart-templates/blob/master/chef-json-parameters-linux-vm/azuredeploy.json#L69.

Please try out replacing new lines with `\n`. It will work.
Hardcoding is not recommended and the quickstart templates shouldn't have it since it's referred by several people.